Cargo.toml

```diff
@@ -65,14 +65,14 @@ bytes = "1.4"
 chrono = { version = "0.4.34", default-features = false }
 ctor = "0.2.0"
 dashmap = "5.4.0"
-datafusion = { path = "datafusion/core", version = "36.0.0" }
-datafusion-common = { path = "datafusion/common", version = "36.0.0" }
+datafusion = { path = "datafusion/core", version = "36.0.0", default-features = false }
```

Comment:
I wonder if we could add some comments (from the PR description) about why we need to use default-features=false here, and for only some of the crates

I am thinking about someone in the future who wants to make a new crate. How do they know if they need to set default features or not 🤔

Reply:
Good point, added a comment